### DISCUSSION TITLE
getSaveContent.extraProps hook for core/post-title doesn't add new attributes to the frontend

### DISCUSSION THREAD

**Discussion by tomektomczuk** [WordPress/gutenberg#45745] [Q&A] [answered]
I have a hook that add extra data attributes to some of my allowed blocks. The problems is the data attributes are not showing on the frontend for post-title block. In the editor everything is fine but not on the save function. I'm assuming that this hooks will not work for save function that is created dynamically. How can I pass these attributes to my allowedblocks inc. core/post-title ?

```
const addMarginPaddingDataAttributesToFrontEnd = (
    props,
    block,
    attributes
) => {
    if ( ! allowedBlocks.includes( block.name ) ) {
        return props;
    }

    const margins = prepareDataString( attributes.margins );
    const paddings = prepareDataString( attributes.paddings );

    const newAttributes = {};
    if ( margins !== '' ) {
        newAttributes[ 'data-margin' ] = margins;
    }

    if ( paddings !== '' ) {
        newAttributes[ 'data-padding' ] = paddings;
    }
    return assign( props, newAttributes );
};
addFilter(
    'blocks.getSaveContent.extraProps',
    'custom/save-data-attributes',
    addMarginPaddingDataAttributesToFrontEnd
);
```
<img width="2140" alt="Screenshot 2022-11-13 at 15 43 47" src="https://user-images.githubusercontent.com/40096666/201533134-d20a6b5b-53f7-4001-aee2-f7a139aa2aaa.png">
<img width="2077" alt="Screenshot 2022-11-13 at 15 46 02" src="https://user-images.githubusercontent.com/40096666/201533138-28f816d0-1ea4-415c-b010-4b2d55c2f785.png">

**Comment by t-hamano** [comment]
My guess is that for dynamic blocks, the rendering needs to be filtered and explicitly given attributes.

Either [render_block](https://developer.wordpress.org/reference/functions/render_block/) or [render_block_{$this->name}](https://developer.wordpress.org/reference/hooks/render_block_this-name/) would work.

**Comment by tomektomczuk** [comment, marked as answer]
For anyone looking for a solution, we have used PHP DOMDocument class.
